controllers/topology/reconcile_state.go

```diff
@@ -229,6 +229,79 @@ func (r *ClusterReconciler) reconcileControlPlane(ctx context.Context, s *scope.
 		return errors.Wrapf(err, "failed to update %s", tlog.KObj{Obj: s.Desired.ControlPlane.Object})
 	}
 
+	// If the Control has defined a MachineHealthCheck set the ownerReferences and reconcile it.
+	if s.Desired.ControlPlane.MachineHealthCheck != nil || s.Current.ControlPlane.MachineHealthCheck != nil {
+		var owner *unstructured.Unstructured
+		var err error
+
+		// Get the recently created ControlPlane Object to use as an ownerReference for the MachineHealthCheck.
+		if owner, err = r.getReference(ctx, &corev1.ObjectReference{
+			APIVersion: s.Desired.ControlPlane.Object.GetAPIVersion(),
+			Kind:       s.Desired.ControlPlane.Object.GetKind(),
+			Name:       s.Desired.ControlPlane.Object.GetName(),
+			Namespace:  s.Desired.ControlPlane.Object.GetNamespace(),
+		}); err != nil {
+			return errors.Wrapf(err, "failed to retrieve controlPlane ownerReferences for %s", tlog.KObj{Obj: s.Desired.ControlPlane.MachineHealthCheck})
+		}
+
+		// Set the ControlPlane Object and the Cluster as owners for the MachineHealthCheck
+		s.Desired.ControlPlane.MachineHealthCheck.SetOwnerReferences([]metav1.OwnerReference{
+			*ownerReferenceTo(owner),
+			*ownerReferenceTo(s.Current.Cluster),
+		})
```

Comment:
Let's make this call only when the MHC does not exists, so we avoid a Get request in all the following reconcile